File: src/main/java/data/SerializationData.java
package data;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import domain.stock.Holding;
import domain.stock.History;

public class SerializationData {
    public final Map<String, List<History>> stockHistory;
    public final Map<String, List<Holding>>  holdings;

    public SerializationData(final Map<String, List<History>> stockHistory, final Map<String, List<Holding>> holdings) {
        this.stockHistory   = new HashMap<>(stockHistory);
        this.holdings       = new HashMap<>(holdings);
    }
}

File: src/main/kotlin/data/SerializationData.kt
package data

import domain.assets.security.SecurityHistory
import domain.assets.security.SecurityHolding
import kotlin.time.Instant

@Deprecated("This only used for old serialized data")
data class SerializationData(
    val stockHistory: Map<String, List<SecurityHistory>>,
    val holdings: Map<String, List<SecurityHolding>>
)

data class SecuritySerializationData(
    val identifier: Identifier,
    val history: List<MarketHistory>
) {
    data class Identifier(
        val isin: String,
        val exchange: String,
        val tickerSymbol: String,
    )

    data class MarketHistory(
        val date: Instant,
        val closingPrice: Double,
    )
}